wuvt/auth/auth_manager.py

```diff
@@ -0,0 +1,119 @@
+from flask import abort, make_response, session, url_for
+from flask_login import current_user, LoginManager
+from flask_oidc import OpenIDConnect
+from functools import wraps
+
+from wuvt.models import User
+from .blueprint import bp
+
+
+class AuthManager(object):
+    def __init__(self, app=None):
+        self.bp = bp
+        self.all_roles = [
+            'admin',
+            'content',
+            'business',
+            'library',
+            'missioncontrol',
+        ]
+
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app):
+        from . import views
+
+        self.app = app
+
+        app.config.setdefault('AUTH_METHOD', 'local')
+        app.config.setdefault('AUTH_ROLE_GROUPS', {
+            'admin': ['webmasters'],
+            'content': ['webmasters'],
+            'business': ['business'],
+            'library': ['librarians'],
+            'missioncontrol': ['missioncontrol'],
+        })
+
+        # deprecated
+        if app.config.get('LDAP_AUTH', False):
+            app.config['AUTH_METHOD'] = 'ldap'
+
+            # support old-style LDAP role map
+            for role in self.all_roles:
+                value = app.config.get('LDAP_GROUPS_{}'.format(role.upper()),
+                                       None)
+                if value is not None:
+                    app.config['AUTH_ROLE_GROUPS'][role] = value
+
+        self.login_manager = LoginManager()
+        self.login_manager.login_view = "auth.login"
+        self.login_manager.init_app(app)
+        self.login_manager.user_loader(load_user)
+
+        if app.config['AUTH_METHOD'] == 'oidc':
+            import flask_oidc
+            flask_oidc.logger = app.logger
+
+            with app.app_context():
+                app.config.update({
+                   'OIDC_SCOPES': ['openid', 'profile', 'email', 'groups'],
```

Comment:
The groups scope is required because of how we handle authentication. I can't think of a situation where we'd need to have customization of this, but if you can come up with one we can evaluate the best way to go from there.